### PR TITLE
Promote NonPreempt feature gate to beta

### PR DESCRIPTION
**Pull request by denkensk**
**What type of PR is this?**
/kind feature
/priority important-soon
/milestone 1.19

**What this PR does / why we need it**:
Promote NonPreempt to Beta, Update NonPreempt feature gate to beta

**Which issue(s) this PR fixes**:
Ref https://github.com/kubernetes/enhancements/issues/902

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/api/scheduling/v1/types.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ type PriorityClass struct {
 	// PreemptionPolicy is the Policy for preempting pods with lower priority.
 	// One of Never, PreemptLowerPriority.
 	// Defaults to PreemptLowerPriority if unset.
-	// This field is alpha-level and is only honored by servers that enable the NonPreemptingPriority feature.
+	// This field is beta-level.
```
Comment (liggitt) [inline staging/src/k8s.io/api/scheduling/v1/types.go:57]:
```suggestion
	// This field is beta-level, gated by the NonPreemptingPriority feature-gate.
```

**Comment by denkensk** [comment]
@liggitt  Please take a look again ~~ thanks

File: staging/src/k8s.io/api/scheduling/v1/types.go (hunk 1 above)
Comment (denkensk) [reply staging/src/k8s.io/api/scheduling/v1/types.go]: Done

**Comment by tengqm** [comment]
This needs a follow up for updating the docs site.

**Comment by denkensk** [comment]
https://github.com/kubernetes/website/pull/21453 @tengqm  Ready for Review

**Pull request merged**